src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -742,14 +757,315 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	//#region Screenshots
 
-	async getScreenshot(windowId: number | undefined, options?: INativeHostOptions): Promise<VSBuffer | undefined> {
+	async getScreenshot(windowId: number | undefined, rect?: IRectangle, options?: INativeHostOptions): Promise<VSBuffer | undefined> {
 		const window = this.windowById(options?.targetWindowId, windowId);
-		const captured = await window?.win?.webContents.capturePage();
+		const captured = await window?.win?.webContents.capturePage(rect);
 
 		const buf = captured?.toJPEG(95);
 		return buf && VSBuffer.wrap(buf);
 	}
 
+	async getElementData(windowId: number | undefined, offsetX: number = 0, offsetY: number = 0, token?: CancellationToken): Promise<IElementData | undefined> {
```

Comment:
What if this takes a `URI` of the webview to target instead. To start picking, you would read the current URI (or maybe origin?) of the active (or last-active) webview from the `IWebviewService` and then find the target matching that URI here. Then you would not be tied to requiring the simple browser.

Reply:
yeah that's a good idea. i think will maybe defer to next iteration and try this out